### ISSUE TITLE
CheckConstraint not created when creating table from class

### ISSUE THREAD

**Issue by mvolfik** [sqlalchemy/sqlalchemy#5035]
```python
class Thing(Base):
        __tablename__ = 'things'
        id = sa.Column(sa.Integer, primary_key=True)
        number = sa.Column(sa.Integer)
        # doesn't work, but should according to https://docs.sqlalchemy.org/en/13/core/constraints.html#check-constraint
        sa.CheckConstraint("number>5")
        # works, but I couldn't find it in docs, only some random example site
        __table_args__ = (sa.CheckConstraint("number>5"),)
```

**Comment by zzzeek** [comment]
declarative table configuration is documented at:

https://docs.sqlalchemy.org/en/13/orm/extensions/declarative/table_config.html

hope this helps!

**Comment by zzzeek** [comment]
also the document at https://docs.sqlalchemy.org/en/13/core/constraints.html#check-constraint refers to a core Table construct, not a mapped class.     the ORM and Core tutorials should be visited to set up a little bit of the basic terminology.

**Comment by mvolfik** [comment]
Oh, sorry, I didn't notice that it's the table definition and not the class one in docs. Developing with SQLAlchemy for the first time and in combination with Flask-SQLA, which strongly prefers the ORM way...

So from class is only copied name, columns and metadata? Ok then, thanks for clarification, I guess I should've read the tutorial more carefully....

**Comment by zzzeek** [comment]
i will grant that the `__table_args__` thing is kind of buried.  in future SQLAlchemy releases I hope to have declarative more of the default within the docs and it should be easier to find.

**Comment by ruslaniv** [comment]
It still does not work for me
I have set up `__table_args__` like this:
```
__table_args__ = (
        UniqueConstraint(
            "trader_id",
            "market_id",
            "pocket_id",
            name="settings_trader_market_pocket_uq",
        ),
        CheckConstraint(
            "drawdown_percent > 0 AND drawdown_percent <= 100",
            name="chk_drawdown_percent",
        ),
)
```
but the migration file only contains unique constraint and not the check constraint.

**Comment by CaselIT** [comment]
that's documented in https://alembic.sqlalchemy.org/en/latest/autogenerate.html#what-does-autogenerate-detect-and-what-does-it-not-detect 

> Autogenerate can’t currently, but will eventually detect:
> Some free-standing constraint additions and removals may not be supported, including PRIMARY KEY, EXCLUDE, CHECK; these are not necessarily implemented within the autogenerate detection system and also may not be supported by the supporting SQLAlchemy dialect.

So you will have to add them manually

**Comment by ruslaniv** [comment]
Yes, I can confirm that once I have deleted the table (and the migration file that created that table) I was able to create a new migration file that contained proper table constraints. 

Is there any estimation on when alembic might be able to detect new constraints on existing tables? 
I assume there some major technical difficulties to implement it or it would have been implemented a long time ago?
